<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>873</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>A CS123 Final</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QWidget" name="canvas3D" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QDockWidget" name="BurningTreeDock">
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="burningDockContents">
    <widget class="QCheckBox" name="BurningButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>270</y>
       <width>100</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Burning</string>
     </property>
    </widget>
    <widget class="QLabel" name="burningTreeLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>240</y>
       <width>211</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Let's make this cactus burn!</string>
     </property>
    </widget>
    <widget class="QLabel" name="CactusControlsLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Cactus Controls</string>
     </property>
    </widget>
    <widget class="QSlider" name="SizeControl">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>50</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="SizeLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>58</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Size</string>
     </property>
    </widget>
    <widget class="QLabel" name="AngleLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Branching Angle</string>
     </property>
    </widget>
    <widget class="QSlider" name="AngleControl">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>90</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="LengthALabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Branch Length</string>
     </property>
    </widget>
    <widget class="QSlider" name="LengthControl">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>130</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="BranchWidthLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Branch Width</string>
     </property>
    </widget>
    <widget class="QSlider" name="BranchWidthControl">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>170</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QCheckBox" name="RandomCactus">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>86</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Random</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="SizeInput">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>50</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="AngleInput">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>90</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="LenInput">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>130</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="BranchInput">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>170</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QWidget" name="cameraDockEmptySpace_2" native="true">
     <property name="geometry">
      <rect>
       <x>6</x>
       <y>605</y>
       <width>230</width>
       <height>175</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>6</x>
       <y>565</y>
       <width>230</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Resize the window to change the aspect ratio.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="cameraAspectRatio_2">
     <property name="geometry">
      <rect>
       <x>6</x>
       <y>541</y>
       <width>230</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Aspect ratio:</string>
     </property>
    </widget>
    <widget class="QWidget" name="cameraFrustum_2" native="true">
     <property name="geometry">
      <rect>
       <x>6</x>
       <y>417</y>
       <width>230</width>
       <height>76</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_6">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <property name="verticalSpacing">
       <number>5</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="cameraFovLabel_2">
        <property name="text">
         <string>FOV</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QLineEdit" name="cameraFovTextbox_2">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSlider" name="cameraFovSlider_2">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>0</height>
         </size>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="cameraNearLabel_2">
        <property name="text">
         <string>Near</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QLineEdit" name="cameraNearTextbox_2">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="cameraFarLabel_2">
        <property name="text">
         <string>Far</string>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QLineEdit" name="cameraFarTextbox_2">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSlider" name="cameraNearSlider_2">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>0</height>
         </size>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSlider" name="cameraFarSlider_2">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>0</height>
         </size>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QPushButton" name="resetSliders_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>503</y>
       <width>242</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Reset FOV / Near / Far</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
